server/src/main/java/org/opensearch/transport/TransportHandshaker.java

```diff
@@ -105,7 +106,11 @@ void handleHandshake(TransportChannel channel, long requestId, StreamInput strea
             throw new IllegalStateException("Handshake request not fully read for requestId [" + requestId + "], action ["
                 + TransportHandshaker.HANDSHAKE_ACTION_NAME + "], available [" + stream.available() + "]; resetting");
         }
-        channel.sendResponse(new HandshakeResponse(this.version));
+        if (stream.getVersion().before(Version.V_1_0_0)) {
```

Comment:
As you have not changed the minCompatibilityVersion for OpenSearch.
Now OpenSearch node will return 7.10.2 for both Elasticsearch & OpenSearch on the other side of the connection. 

So, if we have new OpenSearch cluster with OpenSearch nodes only then also nodes will send 7.10.2 as the version. Is this acceptable? If not, we will have to change the minCompatibilityVersion for OpenSearch to 6.7.99 allows to identify the OpenSearch node on the other side of the connection.

Reply:
The PR doesn't yet address the minCompatibilityVersion which is the next part. Alternatively, we can close that in a separate PR.

Reply:
Merged in [proposed fix](https://github.com/shwetathareja/OpenSearch/commit/63a6a8c431fb8103201548354f0e1b06fa0e8a8e) with a slight change